src/library/Utilities/RemoteContainerConnectionDetailsResolver.cs

```diff
@@ -293,6 +297,15 @@ internal class RemoteContainerConnectionDetailsResolver
                         _log.Warning("{0} pods found, but they are all running RemoteAgent", pods.Count);
                         throw new UserVisibleException(_operationContext, Resources.SpecifiedServiceBackedByMultipleRemoteAgents, service.Metadata?.Name, pods.Count);
                     }
+
+                    // filter pods which are not replicaset or statefulset
+                    podsNotRunningDevHostAgent = podsNotRunningDevHostAgent.Where(p => p.Metadata?.OwnerReferences?.Any(r => StringComparer.OrdinalIgnoreCase.Equals(r.Kind, "ReplicaSet") || StringComparer.OrdinalIgnoreCase.Equals(r.Kind, "StatefulSet")) ?? false);

```

Comment:
💡 for `nullable types` do we care to inform user or this is fine?

Please note: Also how about we enum the string match here for anywhere else if we need to use. like `ReplicaSet` or `StatefulSet` et. al.?

Reply:
Yes of course I can create a Enum for ReplicaSet and StatefulSet. For nullable types ex: what if metadata or owner references is empty ? If they are empty I don't think it is operational or working pod and user would get more errors.

Reply:
You can have a working pod that has no owner property if the pod is created directly, can't you?

Unrelated: I think it'd be nice to simplify the linq expression here by introducing an intermediate function. For example, you could change line 302 to:
```csharp
podsNotRunningDevHostAgent = podsNotRunningDevHostAgent.Where(p => p.IsOwnerOfKind("ReplicaSet", "StatefulSet"));
```
I think having a named function in there makes the intent clearer - what do you reckon? You could put the implementation in `V1podExtensions.cs`, like:
```csharp
        public static bool IsOwnerOfKind(this V1Pod pod, params string[] kinds) =>
            pod.Metadata?.OwnerReferences?.Any(r => kinds.Any(k => StringComparer.OrdinalIgnoreCase.Equals(r.Kind, k))) ?? false;
```

Reply:
updated the code as per your suggestion, but will see how pods without owner references will affect this change.